### PR TITLE
Improve start page when GitHub App not installed

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Should do as described [here](https://github.com/gitpod-io/gitpod/issues/6875#issue-1062916651):
1. Project description text updated (this persists also during loading state)
2. When no account/group/org is selected, the user account and provider is displayed
3. More text updates for when a GitHub app is not installed
4. Remove the org configuration text 

| CHANGES |
| - |
| Screenshot 2021-12-14 at 20 14 00 |

## Related Issue(s)

Relates #6875 

## How to test
1. Go to https://laushinka-start-page-6875.staging.gitpod-dev.com/new
2. Log in with GitHub
3. You should see the start page without GitHub app installed

## Release Notes

```release-note
Improved start page when a GitHub app is not installed.
```

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#6875*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
This is still a WIP. Just to make it easier to collect input from you @jldec @gtsiolis.

**Comment by jldec** [comment]
Thanks @laushinka 
I have updated the wording and added another screenshot in the OP of issue #6875 and below

- Please make the width of this control the same as the current width of https://gitpod.io/new (I don't know why it got narrower in your PR)
- Fix updated wording (top, in grey area, and at bottom)
- Add line of text above the dropdown in 2 variants - for app install, and for repo select
- Put back the "Repository not found? [Reconfigure]()" for repo select on GitHub only - not required for GitLab

**from @gtsiolis**

- Sentence case for Learn more
- Remove the outer border
- Change Authorize to `Configure Gitpod App`
- Check right padding for the avatar

---
Screenshot 2021-12-10 at 11 46 11

**Comment by gtsiolis** [comment]
Haven't dived into the changes yet but just seeing the avatar, username, and provider URL in the dropdown element makes this page 10x better. 🤯 🔝

**Comment by laushinka** [comment]
@jldec Thanks for the review!
> * Please make the width of this control the same as the current width of https://gitpod.io/new (I don't know why it got narrower in your PR)
What are the width differences?
> * Fix updated wording (top, in grey area, and at bottom)
For the top, is it `Projects allow you to manage prebuilds and workspaces for your repository.`?

**Comment by jldec** [comment]
> What are the width differences?

I think what I'm seeing is a width difference between the app install layout and the repo-select layout.
I noticed this particularly when I moved the text "Select Git repository from" text down to be right above the dropdown. In one case it fits nicely, in the other it hangs over the sides.

**Comment by laushinka** [comment]
> I think what I'm seeing is a width difference between the app install layout and the repo-select layout.

@jldec Noted. I'll check with @gtsiolis on which card in Figma I should follow, so I can inspect the attributes. And I will also wait for George's input so I'll tackle them altogether 🙏🏽

**Comment by gtsiolis** [comment]
Go markdown tables[[1](https://github.com/gitpod-io/gitpod/pull/7163#issuecomment-991018203)]! Looking at this now! 👀

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/NewProject.tsx:280]: question: Could this be rephrased to a simpler text? Organizations is also not a thing on GitLab as they are called Groups[[1](https://docs.gitlab.com/ee/user/group/)].

```suggestion
            <p className="text-gray-500 text-center text-base">Projects allow you to manage prebuilds and workspaces for your repository. <a href="https://www.gitpod.io/docs/teams-and-projects" rel="noopener" className="gp-link">Learn more</a></p>
            <p className="text-gray-500 text-center text-base mt-12">Select acccount on <b>{selectedProviderHost}</b> (<a className="gp-link cursor-pointer" onClick={() => setShowGitProviders(true)}>change</a>)</p>
```

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/NewProject.tsx:281]: **issue**: Could we remove the outer border here?

**Review by gtsiolis — APPROVED** [review]
Great work, @laushinka! 🌟 

Left some comments below regarding UX changes. Current changes already improve this area of the product. Feel free to ingore any of these and merge for this iteration. ➿ 

Re-posting from https://github.com/gitpod-io/gitpod/pull/7163/files#issuecomment-990959643:
> Haven't dived into the changes yet but just seeing the avatar, username, and provider URL in the dropdown element makes this page 10x better. 🤯 🔝

Re-apporving to unblock merging but holding in case you'd resolve any of the comments below.

/hold

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7163#pullrequestreview-830708388" title="Approved">gtsiolis</a>*

Associated issue: *#6875*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:280]: Makes sense!

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:285]: @gtsiolis Would you mind recording it? 🙏🏽

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:281]: Thank you for all the CSS pointers, @gtsiolis! I definitely could use them 🙈

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx]: I kept the first line, but the loading state runs when we don't have the provider yet, which the second line needs.

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:289]: Oh I see it now

**Comment by laushinka** [comment]
@gtsiolis Do we have a fallback image for when there is no avatar available for GitLab?

Screenshot 2021-12-14 at 17 16 55

**Comment by laushinka** [comment]
@jankeromnes Would appreciate your review here for the code changes 🤗 The changes themselves are not a lot, but the file _is_ a bit long and can be tidied up in the future, so please let me know if anything is difficult to read.

**Review by gtsiolis — COMMENTED** [review]
Hey @laushinka! Added a few more nitpicks below.

**question**: Also, did we drop the idea of the reconfigure link at the bottom from https://github.com/gitpod-io/gitpod/pull/7163#issuecomment-990914397?

**Comment by gtsiolis** [comment]
Looping in @jldec in case this could seem useful to keep around. Ideally, we could get rid of the change link next to the provider here. 💡

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:293]: Totally better to fix the underlying issue, but thought we would want to have an intermediate thing. But I agree with you to leave it 👏🏼

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [reply components/dashboard/src/projects/NewProject.tsx]: Looks great! 🙂

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [reply components/dashboard/src/projects/NewProject.tsx:293]: Let's cut some corners to ship this! 🧀 

Looping in @jldec in case this (https://github.com/gitpod-io/gitpod/issues/5330) is something we'd like to prioritize soon.

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:287]: > did we drop the idea of the reconfigure link at the bottom from [#7163 (comment)](https://github.com/gitpod-io/gitpod/pull/7163#issuecomment-990914397)?

Thanks for redirecting me to this part that I missed. Will put it back.

**Comment by gtsiolis** [comment]
**thought**: One more thing, possibly out of the scope of this PR but we could probably align the provider selection with these updates here. I've opened https://github.com/gitpod-io/gitpod/issues/7237 to track this. Cc @jldec

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:287]: > Once we change the dropdown height, using `top-2` instead of `top-1/4` could suffice. However, maybe this is something we could also tackle in a next step as it's a minor issue which was already existing before this PR.

@gtsiolis Ah, I see. Would you mind pointing me to where this 42px is set? If it's a straightforward change then I'll do it here, otherwise we can tackle it later.

Comment (gtsiolis) [reply components/dashboard/src/projects/NewProject.tsx:287]: @laushinka I don't think the `42px` height is directly set anywhere as it's because of the padding used from the inner elements of the dropdown.

Screenshot 2021-12-14 at 8 43 58 PM

Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx]: Did a little hack which pushed up the avatar a tiny bit, but still not exactly centered. I'll try to play a bit before this PR closes, but if not then in the next issue 🙏🏽

**Comment by laushinka** [comment]
> **thought**: One more thing, possibly out of the scope of this PR but we could probably align the provider selection with these updates here. I've opened #7237 to track this. Cc @jldec

It looks straightforward and I don't mind working on it here, but the issue says TBD so it's probably better to wait until we are aligned?

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [reply components/dashboard/src/projects/NewProject.tsx]: > I'll try to play a bit before this PR closes, but if not then in the next issue

DEAL 🤝

**Comment by gtsiolis** [comment]
> It looks straightforward and I don't mind working on it here, but the issue says TBD so it's probably better to wait until we are aligned?

Let's keep this out of the scope for now. Only posted this here in case this PR made any changes to that step (page), too. ➿

File: components/dashboard/src/projects/NewProject.tsx
Comment (jldec) [inline components/dashboard/src/projects/NewProject.tsx:337]: This text is a little long and on GitHub we will only trigger **all** branches if you configure that in .gitpod.yml. Let's adjust wording as [spec'ed](https://github.com/gitpod-io/gitpod/issues/6875#issue-1062916651)

`Additional authorization is
required for our GitHub App to
watch your repositories and
trigger prebuilds.`

File: components/dashboard/src/projects/NewProject.tsx
Comment (jldec) [inline components/dashboard/src/projects/NewProject.tsx:349]: This text should only appear when with the repositorylist is showing. It is not required when we prompt for app authorization.

File: components/dashboard/src/projects/NewProject.tsx
Comment (jldec) [inline components/dashboard/src/projects/NewProject.tsx:280]: When the repo list is showing, this text should change to 
`Select a Git repository on xxx. (change)` like we used to have.

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/NewProject.tsx:361]: **Praise**: Many thanks for adding back this text in the loading screen! (This is actually something I broke recently 🙈)

**Thought**: Maybe in this case, the slight duplication is not a problem, but it still feels a little awkward that someone could possibly fix or improve the copy here (or update the link, etc) but accidentally forget to update the other occurrence of the same text (or vice versa). Maybe it's a tiny bit safer to de-duplicate this text again?

Possible solutions could be:
- `const ProjectsDescription = <div>...</div>;`
    - defined somewhere above `renderLoadingState` and `renderRepos`, in the same scope
    - used as `{ProjectsDescription}`
- `const renderProjectsDescription = () => <div>...</div>;`
    - defined somewhere above `renderLoadingState` and `renderRepos`, in the same scope
    - used as `{renderProjectsDescription()}`
- `function ProjectDescription() { return <div>...</div>; }`
    - defined as a separate component, e.g. at the bottom of the file
    - used as `<ProjectDescription/>`

I'm not really sure which one is cleaner / more readable 😅 please feel free to pick the option you prefer, or to ignore this comment entirely for now.

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:349]: And only for GitHub, right? Or all providers?

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/NewProject.tsx:226]: **Praise**: Super cool that we no longer need this secondary/orthogonal authorization flow! 🙌 (I personally found it super confusing to have both "Reconfigure" and "Grant org scope" -- must have been even worse for users who don't actually know what these two things even mean 🙈)

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/NewProject.tsx:244]: **Thought**: This is definitely out of scope for this code review, but I wonder if maybe we shouldn't have the GitHub user in the list all the time, and not just when there is no `selectedAccount`?

The rationale goes like this:

| 1. First time seeing this page | 2. After installing app for an org | 3. After installing app for your user |
| --- | --- | --- |
| Screenshot 2021-12-15 at 11 13 38 | Screenshot 2021-12-15 at 11 14 13 | Screenshot 2021-12-15 at 11 14 47 |
| Cool! I see my user 👍 | Oh no! My user is gone? | Aah, my user is back 😌 |

Adding the user in case 2 (i.e. there is a selected account, but the user is still not in the list of accounts) might help.

I think this could be achieved by replacing the `!selectedAccount` condition by something that checks for whether `user` is actually in the list of `accounts`.

However, this should probably be a follow-up issue / PR. 😇

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/NewProject.tsx:280]: **Nit**: I think the name Git is always capitalized.

**Review by jankeromnes — APPROVED** [review]
Many thanks @laushinka! As mentioned on Slack, this is so much nicer now 😍 ✨

Also, the code looks great to me. 🎯 Many thanks for caring about readability. 💯

I've added a few thoughts, but they're entirely optional or out of scope, so please feel free to read them later. 🌴

Approving because the code can be merged as is and works flawlessly.

/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: a2decbc7e42bfaf6c5af7e1f77989fc5f671d4c9</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7163#pullrequestreview-830708388" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7163#pullrequestreview-832596423" title="LGTM">jankeromnes</a>*

Associated issue: *#6875*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:361]: You're totally right and I actually extracted the logic in the beginning. But TypeScript didn't like me doing that 🥲 as it always considered one of the variables undefined even though I checked. I will try extracting the HTML then and maybe that would work. Completely agree that having this duplication is prone to being overlooked and bugs.

File: components/dashboard/src/projects/NewProject.tsx
Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:244]: You're right - let's discuss this in a follow up PR, as we also have follow-up tasks for the same issue 🙌🏽

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by laushinka** [comment]
@jankeromnes Pushed improvements for the code 🙌🏽

**Comment by laushinka** [comment]
Currently the preview env is broken as I rebased (probably should've held off on that) while we have an [ongoing problem](https://gitpod.slack.com/archives/C021ENP4CEN/p1639564118143300).

**Comment by laushinka** [comment]
[Preview env](https://laushinka-start-page-6875.staging.gitpod-dev.com/new) all good now.

File: components/dashboard/src/projects/NewProject.tsx
Comment (jldec) [reply components/dashboard/src/projects/NewProject.tsx:353]: Yes - GitHub only.

**Comment by jldec** [comment]
/lgtm
Nice result - thanks @laushinka .

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: bc4371704dd9c17a248415caf41c914d484c9323</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7163#pullrequestreview-830708388" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7163#pullrequestreview-832596423" title="LGTM">jankeromnes</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7163#issuecomment-994899644" title="LGTM">jldec</a>*

Associated issue: *#6875*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis,jankeromnes,jldec]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
